services/application.go

```diff
@@ -133,3 +138,28 @@ func (app *ChainlinkApplication) RemoveAdapter(bt *models.BridgeType) error {
 
 	return nil
 }
+
+// ResumeSleptRuns restarts the timer for all of the slept runs.
+func (app *ChainlinkApplication) ResumeSleptRuns() error {
+	pendingRuns, err := app.Store.JobRunsWithStatus(models.RunStatusPendingSleep)
+	if err != nil {
+		return err
+	}
+	for _, run := range pendingRuns {
+		go func() {
+			ExecuteRun(run, app.Store, models.RunResult{})
+		}()
+	}
+	return nil
+}
+
+func (app *ChainlinkApplication) listenToRunChannel() {
+	for input := range app.Store.RunChannel {
+		run, err := app.Store.FindJobRun(input.JobRunID)
+		if err != nil {
+			logger.Warn("Application Run Channel Executor: error finding run", "ID", input.JobRunID)
+			continue
+		}
+		ExecuteRun(run, app.Store, input)
```

Comment:
This seems like a bottleneck, the `HeadTracker` will wake up a new goroutine for each `Pending` job, but this will block on the first job that finishes after a `Sleep`. So all JobSpecs with a `Sleep` somewhere in them will get blocked up here.